src/app/api/crud.py

```diff
@@ -18,11 +14,18 @@ async def get(id: int, table: Table):
     return await database.fetch_one(query=query)
 
 
-async def fetch_all(table: Table):
+async def fetch_all(table: Table, query_filter: Optional[Tuple[str, Any]] = None):
     query = table.select()
+    if isinstance(query_filter, tuple):
+        query = query.where(getattr(table.c, query_filter[0]) == query_filter[1])
     return await database.fetch_all(query=query)
 
 
+async def fetch_one(table: Table, query_filter: Tuple[str, Any]):
+    query = table.select().where(getattr(table.c, query_filter[0]) == query_filter[1])
+    return await database.fetch_one(query=query)
+
```

Comment:
1. Happy to move things out of the route code, you're right
2. I disagree on that point. If things cannot be refactored using this, then we'll leave it as is, but here when I checked the code, it was very similar to a fetch_all only with a filter.
3. On that part, I agree about understandability. But I'm not sure about the class-oriented design of CRUD, your last suggestion could be relevant in that sense.

To reproduce the same as crud classes, we can have a crud folder with a file for each table. I see it as cleaner than using classes only to instantiate them in the same file for importing reasons. 

More generally about refactoring, as we increase the number of routes, I think it's good to minimize the number of LoC to add to have another feature implemented. If we keep this design, I do believe this will become substantial :/

Reply:
Adding a query_filter option is a smart way to avoid code duplication. It's fine for me as a first iteration for the refactor. But as mentioned Florian what if we need something more complex ? I guess a good move would be to have a BaseCRUD with the current operations and each model (UserCRUD, DeviceCRUD, etc.) would inherit from that BaseCRUD to contain special/complex operations called by the route. What do you think @florianriche @frgfm ?

Reply:
To me, I fully agree that if we get more complex operations, we'll need to extra CRUD (either additional functions in the module or submodules for each table). As of now, I didn't see many reasons not to leverage a query_filter to extend the usage of fetch functions.

However, I agree with @florianriche on moving queries out of the route code, I'll come up with a suggestion shortly!

Reply:
Ok for me as long as we agree that we would need to refactor the CRUD if we need more complex  operations.